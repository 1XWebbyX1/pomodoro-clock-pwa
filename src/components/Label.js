import React from 'react'
import '@fortawesome/fontawesome-free/css/all.css'

class Label extends React.Component {
    render() {
      return (
        <div id={this.props.id} class='label'>
        <p class='head'>{this.props.text}</p>
        <hr class='line' />
        <div class='inline'>
        <i id='inc' class='fa fa-plus inc' onClick={this.props.increment}></i>
        <p class='length'>{this.props.length}</p>
        <i  class='fa fa-minus dec' onClick={this.props.decrement}></i>
        </div>
      </div>
      );
    }
}

export default Label;
