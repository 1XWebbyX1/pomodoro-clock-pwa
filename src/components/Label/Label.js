import React from 'react'
import '@fortawesome/fontawesome-free/css/all.css'

class Label extends React.Component {
    render() {
      return (
        <div id={this.props.id} class='label'>
        <p class='head'>{this.props.text}</p>
        <hr class='line' />
        <div class='inline'>
        <div className='wrap-plus inc' onClick={this.props.increment}><i class='fa fa-plus'></i></div>
        <p class='length'>{this.props.length}</p>
        <div className='wrap-minus dec' onClick={this.props.decrement}><i  class='fa fa-minus'></i></div>
        </div>
      </div>
      );
    }
}

export default Label;
