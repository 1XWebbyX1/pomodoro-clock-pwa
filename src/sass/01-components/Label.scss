.label{
  width: calc(13em + 3vmax);
  height: calc(6.5em);
  color: #a6a6a6;
  background: rgba(0,0,0, 0.3) !important;

  .head{
    top: 10%;
    text-align: center;
    cursor: default;
  }

  .line{
    height: 2px;
    border: none;
    width: calc(13em + 3vmax);
    top: 32%;
    font-size: calc(0.7em + 0.3vmax);
    transform: translateY(-50%);
    background-image: linear-gradient(to bottom, #063f35, #013235, #0d252d, #12181f, #0b0b0c);
  }


  .inline{
  //  margin-top: 3%;
    display: flex;
    justify-content: space-around;
    align-items: flex-start;
    .length{
       font-size: calc(1.2em + 0.3vmax);
        cursor: default;
        margin-top:  -2%;
    }

    .inc{
      color: #FF6D00;
      cursor: pointer;
    }

    .dec{
      color: #FF6D00;
      cursor: pointer;
    }
  }

}
