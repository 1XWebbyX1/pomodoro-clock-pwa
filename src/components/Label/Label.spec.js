// External Depedencies
import React from 'react';
import { shallow, mount, configure } from 'enzyme';
// Our Dependencies
import { expect } from '../utils/chai';
import Adapter from 'enzyme-adapter-react-16';
import renderer from "react-test-renderer";
// Our Component
import Label from './Label';


configure({ adapter: new Adapter() });

describe('Label', () => {
  const incSession = () => {};
  const decSession = () => {});

  it("renders correctly", () => {
     const output = renderer.create(
        <Label id='session-label' text='SESSION LENGTH' length={24} increment={incSession} decrement={decSession}/>
   ).toJSON();
     expect(output).to.matchSnapshot(); //this will confirm content received through props
  });

})
