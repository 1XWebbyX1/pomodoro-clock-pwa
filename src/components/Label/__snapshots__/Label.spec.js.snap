// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<div
  className="label"
  id="session-label"
>
  <p
    className="head"
  >
    SESSION LENGTH
  </p>
  <hr
    className="line"
  />
  <div
    className="inline"
  >
    <div
      className="wrap-plus inc"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <p
      className="length"
    >
      24
    </p>
    <div
      className="wrap-minus dec"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-minus fa-w-14 "
        data-icon="minus"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
